src/m365/spo/commands/folder/folder-list.ts

```diff
@@ -73,30 +86,100 @@ class SpoFolderListCommand extends SpoCommand {
     }
 
     try {
-      const resp = await this.getFolderList(args.options.webUrl, args.options.parentFolderUrl, args.options.recursive);
+      const folderProperties = await this.getItemCount(args.options.parentFolderUrl, args);
```

Comment:
Just like with `spo file list` let's remove this REST call and just issue requests to `/Folders`, checking the nr of returned items to see if we need to execute another request.